project/Bichen07/README.rst

```diff
@@ -0,0 +1,126 @@
+
+Voronoi Diagrams have applications in various fields,
+such as spatial analysis, nearest-neighbor searches, and computational physics.
+
+*****************
+Problem to Solve
+*****************
+Voronoi Diagrams are used in diverse fields ranging from geography and robotics to machine learning. 
+Calculating Voronoi diagrams efficiently is crucial in large-scale simulations and real-time applications.
+
+However, creating Voronoi diagrams in Python using libraries like SciPy can become inefficient when applied to large datasets 
+or when extended to 3D or higher dimensions. C++ is more suited for these heavy computational tasks but lacks a flexible user interface like Python.
+
+The main problem VoronoiHybrid addresses is combining C++'s computational speed for constructing Voronoi diagrams with Python's ease of use 
+and data visualization libraries. We aim to achieve this without data duplication between the two languages, ensuring fast, efficient, and memory-optimal operations.
+
+*****************
+Prospective Users
+*****************
+The users of VoronoiHybrid will primarily be researchers, engineers, and developers in the following fields:
+
+* **Geospatial Analysis**: For dividing geographical regions, territory mapping, and urban planning.
+* **Robotics**: For tasks such as path planning and sensor network design.
+* **Physics Simulations**: Modeling atomic structures or simulating particle interactions.
+* **Machine Learning**: For clustering, nearest-neighbor searches, and boundary definitions.
+
+*****************
+System Architecture
+*****************
+The VoronoiHybrid system will be composed of two main parts:
+
+* **Core Voronoi Computation (C++11)**: 
+Efficient computation of the Voronoi diagram will be done in C++ 
+using a combination of computational geometry algorithms (Fortune’s sweep line algorithm or Delaunay triangulation). 
+The C++ implementation will manage memory efficiently and handle large datasets.
+
+* **Python Integration (pybind11)**: 
+I will provide a Python interface for ease of use and integration with Python-based libraries 
+such as Matplotlib for visualization and NumPy for data manipulation. 
+This interface will allow users to compute Voronoi diagrams from Python without significant performance penalties or memory duplication.
+
+
+*****************
+API Description
+*****************
+
+Function in Python
+++++++++++++++++++
+* ``compute()``: Constructs the Voronoi diagram and returns vertices and edges. 
+
+* ``get_cells()``: Returns the Voronoi cells, each corresponding to a seed point. 
+
+* ``get_edges()``: Returns edges of the diagram, useful for visualization. 
+
+* ``get_vertices()``: Returns the vertices at the intersection of the edges.
+
+
+*****************
+Engineering Infrastructure
+*****************
+The project will use:
+
+* Unit Testing: C++ unit tests will be created using Google Test for validating the core computation of the Voronoi diagrams. 

```

Comment:
FYI, you may or may not need to use Google-testing for C++.  It depends on how you architecture the code.